internal/pkg/instrumentation/manager.go

```diff
@@ -54,6 +58,15 @@ type Manager struct {
 	probeMu         sync.Mutex
 	state           managerState
 	stateMu         sync.RWMutex
+	collectionOpts  *ebpf.CollectionOptions
+}
+
+// ProbeReference is used by the Manager to track an initialized reference
+// to a Probe and its related resources such as its ebpf.Collection and io.Closers.
+type ProbeReference struct {
```

Comment:
This can be unexported. It is only used internal to this package.

Reply:
It's also not clear why this is needed. The collection field is never used (it is ineffectively set, but never read), and the `closers` field could just be captured in a thin wrapper around a `Probe` implementation (given it is already an interface).

Reply:
replied above that this should have been set on a pointer, sorry about that confusion. But I'm curious what you mean by the closers field being a wrapper around a Probe. If I understand you correctly, that sounds like what I'm basically trying to do here, but could you explain what you have in mind? If it simplifies this then awesome